### ISSUE TITLE
[Bug] Data races between initDecoder and initEncoder operations

### ISSUE THREAD

**Issue by sunadoi** [goccy/go-json#529]
In v0.10.4, A data race condition has been detected when running tests with `-race` flag.

![スクリーンショット 2024-12-17 22 31 41](https://github.com/user-attachments/assets/130a4224-9d9a-48b5-8113-b9fd958f59bb)

I suspect the following is occurring from this PR (#490 ): 
- `runtime.AnalyzeTypeAddr()` is being called in both `initDecoder` and `initEncoder`.
- Although using `sync.Once`, since `initEncoderOnce` and `initDecoderOnce` are separate variables, concurrent calls to `initEncoder` and `initDecoder` result in race conditions in `runtime.AnalyzeTypeAddr()` operations.

In our project, tests that simultaneously use `json.Marshal` and `json.Unmarshal` have become flaky.

**Comment by vincentserpoul** [comment]
I also have the same issue

**Comment by lrweck** [comment]
same

**Comment by goccy** [comment]
@toddtreece It seems that your PR caused a regression. I'm currently unable to allocate time to address this, so could you please handle it ?

**Comment by toddtreece** [comment]
@goccy yes, sorry about missing that. i will open a PR to address this tomorrow morning
